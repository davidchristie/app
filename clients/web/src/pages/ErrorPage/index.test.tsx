import { getByText, render, screen } from "../../testing";
import { ErrorPage } from ".";

describe("ErrorPage", () => {
  const title = "Something went wrong";

  it("renders correctly", () => {
    const { container } = render(
      <ErrorPage error={new Error()} resetErrorBoundary={() => {}} />
    );
    const element = screen.getByTestId("ErrorPage");
    expect(getByText(element, title)).toBeDefined();
    expect(element).toBe(container.firstChild);
    expect(element).toMatchSnapshot();
  });

  describe("when 'Retry' button is clicked", () => {
    it("reloads the page", () => {
      const resetErrorBoundary = jest.fn();
      render(
        <ErrorPage
          error={new Error()}
          resetErrorBoundary={resetErrorBoundary}
        />
      );
      const retryButton = screen.getByTestId("ErrorPage__retryButton");
      retryButton.click();
      expect(resetErrorBoundary).toBeCalledTimes(1);
    });
  });
});
