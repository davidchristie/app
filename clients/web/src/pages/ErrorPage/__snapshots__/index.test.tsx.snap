// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage renders correctly 1`] = `
<div
  class="MuiBox-root MuiBox-root-1"
  data-testid="ErrorPage"
>
  <div
    class="MuiBox-root MuiBox-root-2"
    data-testid="Page"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
      data-testid="MainContent"
    >
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
          data-testid="PageHeading"
        >
          Something went wrong
        </h2>
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained"
            data-testid="ErrorPage__retryButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Retry
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
